### ISSUE TITLE
Device names are still saved somewhere, even after account reset

### ISSUE THREAD

**Issue by cweagans** [keybase/client#2026]
I recently reset my account and started over with a new GPG key. When I ran `keybase login`, I was asked for a device name. I used `cweagans-mbp` before, but even after an account reset, I was not able to do that again. Additionally, there's no way to see these device names in the UI or from the CLI.

Rough steps to reproduce:

1. Login to Keybase through the CLI and set a name for a machine (say, `test-machine`).
2. `keybase deprovision`
3. Reset your Keybase account
4. `keybase login` and attempt to use the same machine name.

This may be intentional (undocumented feature), but it seems strange that the account reset feature is described as a sledgehammer approach, but doesn't actually delete all the things. :)

**Comment by cjb** [comment]
Yes, intentional. It's important for security that a {username, device name} pair always refers to the same device.

**Comment by zong0** [comment]
Then perhaps internally device name should be an auto-generated hash, or such a hash added to the user-defined name.

**Comment by mcstafford-git** [comment]
> It's important for security that a {username, device name} pair always refers to the same device

This makes sense to disallow a mimicry or other trickery within a single account provisioning.

Could devices establish a connection between themselves with across an account reset? If not, then perhaps the pairing restriction should be reset along with the rest of the account.
